tests/phpunit/Metadata/Builder/BaseBuilderTest.php

```diff
@@ -0,0 +1,496 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\DbExtractor\Tests\Metadata\Builder;
+
+use RuntimeException;
+use LogicException;
+use TypeError;
+use Keboola\DbExtractor\Exception\InvalidArgumentException;
+use Keboola\DbExtractor\Exception\PropertyNotSetException;
+use PHPUnit\Framework\Assert;
+use Keboola\DbExtractor\Metadata\Builder\Builder;
+use Keboola\DbExtractor\ValueObject;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * This class tests that values set by a Builder set* methods
+ * match values returned by a ValueObject has* and get* methods.
+ */
+abstract class BaseBuilderTest extends TestCase
```

Comment:
Otestovat vsetky set/get/has metody v `Builder` a `ValueObject`- objektoch by bolo strasne vela neprehladneho kodu. Preto je tu tento `BaseBuilderTest` a podobne aj `BaseValueObjectTest`, ktory otestuje vsetky null/not-null hodnoty, ... vsetky hodnoty kde nie je povoleny prazdny retazec  ... vsetky `set`, `has` a `get` metody.